packages/flutter/lib/src/painting/image_stream.dart

```diff
@@ -520,19 +568,22 @@ class MultiFrameImageStreamCompleter extends ImageStreamCompleter {
     _framesEmitted += 1;
   }
 
-  bool get _hasActiveListeners => _listeners.isNotEmpty;
+  int _listenerCount = 0;
+  bool get _hasActiveListeners => _listenerCount > 0;
 
   @override
   void addListener(ImageListener listener, { ImageErrorListener onError }) {
     if (!_hasActiveListeners && _codec != null) {
       _decodeNextFrameAndSchedule();
     }
+    _listenerCount += 1;
     super.addListener(listener, onError: onError);
   }
 
   @override
   void removeListener(ImageListener listener) {
     super.removeListener(listener);
+    _listenerCount -= 1;
```

Comment:
actually this still won't work because if you pass in something that wasn't registered, it will be a no-op.